docs/DafnyRef/Statements.md

```diff
@@ -1027,26 +1063,48 @@ to the right of `=>` for that guard are executed. The statement requires
 at least one of the guards to evaluate to `true` (that is, `if-case`
 statements must be exhaustive: the guards must cover all cases).
 
+In the if-with-cases, a seeqneuce of statements may follow the `=>`; it
+need not be a block statement. Also the sequence of cases may be enclosed in 
+braces but need not be.
+
 The form that used `...` (a refinement feature) as the guard is deprecated.
 
-## 20.12. While Statement {#sec-while-statement}
-````grammar
-WhileStmt =
-  "while"
-  ( LoopSpec
-    AlternativeBlock(allowBindingGuards: false)
-  | Guard
-    LoopSpec
-    ( BlockStmt
-    | /* go body-less */
-    )
-  )
-````
+## 20.12. While Statement ([grammar](#g-while-statement)) {#sec-while-statement}
+
+Examples:
+<!-- %check-resolve -->
+```dafny
+method m() {
+  var i := 10;
+  while 0 < i
+    invariant 0 <= i <= 10;
+    decreases i;
+  {
+    i := i-1;
+  }
+  while * {}
+  i := *;
+  while 
+     decreases if i < 0 then -i else i
+  {
+     case i < 0 => i := i + 1;
+     case i > 0 => i := i - 1;
+  }
```

Comment:
I had looked at the grammar before but did not realized the full shape of while -case loops, very interesting !

Reply:
OK